packages/simple-hub/src/wallet/index.ts

```diff
@@ -0,0 +1,23 @@
+import {Message as WireMessage} from '@statechannels/wire-format';
+import * as R from 'ramda';
+import {signState} from '@statechannels/xstate-wallet/lib/src/store/state-utils';
+import {deserializeMessage} from '@statechannels/xstate-wallet/lib/src/serde/wire-format/deserialize';
+import {serializeMessage} from '@statechannels/xstate-wallet/lib/src/serde/wire-format/serialize';
+import {cHubStateChannelPK, cHubStateChannelAddress} from '../constants';
+import {SignedState, Message, Participant} from '@statechannels/xstate-wallet/lib/src/store/types';
+
+function broadcastRecipents(participants: Participant[]): Participant[] {
+  return participants.filter(p => p.participantId !== cHubStateChannelAddress);
+}
+
+export function respondToMessage(wireMessage: WireMessage): WireMessage[] {
+  const message = deserializeMessage(wireMessage);
+  const lastState = R.last(message.signedStates);
```

Comment:
Why should this only sign the last state?

Should this check that the hub is a participant in the channel?

Reply:
It's a good idea to check if the hub is a participant. Which states should the hub sign? All states?

Reply:
From @andrewgordstewart in Slack: "any state in a channel with the hub"

Reply:
c2769939fd2c9da8925c9d9a4210c49cf21955b9